ipwb/indexer.py

```diff
@@ -170,6 +170,83 @@ def indexFileAt(warcPaths, encryptionKey=None,
         print('\n'.join(cdxjLines))
 
 
+def generateCDXJMetadata(cdxjLines=None):
+    metadata = ['!context ["http://tools.ietf.org/html/rfc7089"]']
+    metaVals = {
+        'generator': "InterPlanetary Wayback v.{0}".format(ipwbVersion),
+        'created_at': '{0}'.format(datetime.datetime.now().isoformat())
+    }
+    metaVals = '!meta {0}'.format(json.dumps(metaVals))
+    metadata.append(metaVals)
+
+    return metadata
+
+
+def joinCDXJFiles(cdxjPath1, cdxjPath2, outputFilePath):
```

Comment:
Do we really need this new `joinCDXJFiles()` function? Can't we just call existing `indexFileAt()` function instead?

Reply:
`indexFileAt` uses a WARC to generate a CDXJ, so no. `joinCDXJFiles()` takes two CDXJs as sources and merges them.

Reply:
I think `joinCDXJFiles()` is redundant here, the `indexFileAt()` already performs merging when an outfile is specified and the outfile already exists. The function first reads whatever is in the outfile before indexing any new content. Then later truncates the file and writes the aggregated output to it. I remember implementing this functionality back when I added the outfile support.

Reply:
`indexFileAt` expects a WARC path as a parameter. It also considers encryption and other aspects. We need a function that, given two CDXJ files, merge them. Perhaps the logic in `indexFileAt()` can reference this new function.

Reply:
No, we do not need to create a temporary incremental CDXJ file and a temporary combine CDXJ file and then reassign the index file to be used by the replay and piling up the throw away old CDXJ file and the incremental CDXJ file. A lot of this code can away by simply reusing what was already there. I will comment in the replay file where we need to make changes.